README-dev.md

```diff
@@ -0,0 +1,177 @@
+
+Alive provides a language extension to support Common Lisp.
+Basic understanding of that language and the Lisp REPL is assumed.
+The specific Common Lisp implementation is Steel Bank Common Lisp.
+
+The Alive codebase is hosted on
+[github](https://github.com/nobody-famous/alive).
+Basic understanding of `git` and `github` usage
+including branches and forking is necessary.
+
+### VSCode Extension Development
+
+Developing extensions for VSCode is quite involved.
+Start learning with the
+[Extension API](https://code.visualstudio.com/api) page.
+
+## Startup
+
+### Install Tools
+
+Install the following tools:
+* Steel Bank Common Lisp (SBCL)
+* JavaScript
+* TypeScript
+
+You should be able to execute the following programs:
+* `git` (or whatever tool you prefer)
+* `sbcl` Common Lisp REPL
+* `npm` JavaScript package manager
+* `tsc` TypeScript compiler
+
+The rest of this document assumes the command-line use of `git`.
+You may prefer a different tool (including VSCode).
+
+### Acquiring the Code
+
+Alive development is done on a fork of the repository.
+Pull Requests (PRs) are made from a branch in that fork
+into the Alive repository.
+
+* [Create a fork](https://docs.github.com/en/get-started/quickstart/fork-a-repo)
+  of the Alive code into your own `github` account.
+* [Clone your fork](https://docs.github.com/en/repositories/creating-and-managing-repositories/cloning-a-repository)
+  onto your development machine.
+
+### Inital Steps
+
+After changing directory into the cloned development directory there are a few first steps.
+
+#### Install JavaScript Modules
+
+Run the command
+```
+npm install
+```
+in order to acquire necessary JavaScript modules
+specified in the `package.json` file.
+
+Failing to do this will result in hundreds of errors
+as the Alive code will not be able to link to
+the VSCode Extension API.
+
+A side effect of this step is the possible incrementing of a version number in the `package-lock.json` file.
+This is an unnecessary change and should not be propagated into the code.
+
+Execute
+```
+git status
+```
+to check for any changes.
+If there are none skip to **Compile Code** below.
+
+Execute
+```
+git diff package-lock.json
+```
+to make sure that the version number is the _only_ thing incremented.
+The `diff` should look something like
+```
+{
+     "name": "alive",
+-    "version": "0.3.19",
++    "version": "0.3.20",
+     "lockfileVersion": 2,
+     "requires": true,
+     "packages": {
+         "": {
+             "name": "alive",
+-            "version": "0.3.19",
++            "version": "0.3.20",
+             "dependencies": {
+                 "axios": "^0.21.4",
+                 "node-stream-zip": "^1.15.0",
+```
+though the version numbers may be different.
+
+If these are the _only_ changes run
+```
+git checkout package-lock.json
+```
+to revert the changes.
+
+_If there are other changes_ then something bad happened.
+It's probably better to not revert the change to `package-lock.json`.
```

Comment:
I'd take out the package-lock stuff. It works just fine to delete it and let npm create a new one. It doesn't really matter what the contents of it are.

Reply:
Removed that section.